### PR TITLE
Change the api LB health probes from TCP to HTTPS 

### PR DESCRIPTION
**Pull request by asalkeld**
ref https://github.com/Azure/ARO-RP/issues/620

### CODE REVIEW

**Comment by jim-minter** [comment]
I'm not marking this as fixing #620 because we need to retrofit this to existing clusters as well.

File: pkg/install/fixlbprobeconfig.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+func (i *Installer) fixLBProbeConfig(ctx context.Context, resourceGroup, lbName, probeName string) error {
+	lb, err := i.loadbalancers.Get(ctx, resourceGroup, lbName, "")
+	if err != nil {
+		return err
+	}
+
+	if lb.LoadBalancerPropertiesFormat.Probes == nil {
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (asalkeld) [reply pkg/install/fixlbprobeconfig.go:23]: https://github.com/Azure/ARO-RP/blob/master/pkg/install/loadbalancer.go#L87
if visibility != api.VisibilityPublic then the public LB won't have any Probes.

[the remark this answers is not in the document]
Comment (jim-minter) [reply pkg/install/fixlbprobeconfig.go:23]: My bad - thanks! Please can you add a comment?

**Comment by mjudeikis** [comment]
Please do not merge this yet. I was poking installer code and I think I found why this was not working for us when using https probes on install.

**Pull request closed without merging**